src/NexusMods.App.UI/Pages/CollectionDownload/CollectionDownloadViewModel.cs

```diff
@@ -130,39 +139,59 @@ public CollectionDownloadViewModel(
+
+            var numDownloadedOptionalItemsObservable = Observable
+                .Return(_revision)
+                .OffUi()
+                .SelectMany(revision => collectionDownloader.DownloadedItemCountObservable(revision, itemType: CollectionDownloader.ItemType.Optional));
 
             var isPremiumObservable = loginManager.IsPremiumObservable;
             var isCollectionInstalledObservable = collectionDownloader.IsCollectionInstalled(_revision);
 
-            numDownloadedObservable.CombineLatest(isPremiumObservable, isCollectionInstalledObservable)
+            numDownloadedRequiredItemsObservable.CombineLatest(isPremiumObservable, isCollectionInstalledObservable)
                 .OnUI()
                 .Subscribe(tuple =>
                 {
-                    var (numDownloaded, isPremium, isCollectionInstalled) = tuple;
-                    var hasDownloaded = numDownloaded == RequiredDownloadsCount;
+                    var (numDownloadedRequiredItems, isPremium, isCollectionInstalled) = tuple;
+                    var hasDownloadedAllRequiredItems = numDownloadedRequiredItems == RequiredDownloadsCount;
 
-                    _canInstall.OnNext(!isCollectionInstalled && hasDownloaded);
-                    _canDownload.OnNext(!hasDownloaded && isPremium);
+                    CountDownloadedRequiredItems = numDownloadedRequiredItems;
+                    _canInstallRequiredItems.OnNext(!isCollectionInstalled && hasDownloadedAllRequiredItems);
+                    _canDownloadRequiredItems.OnNext(!hasDownloadedAllRequiredItems && isPremium);
 
-                    if (hasDownloaded)
+                    if (hasDownloadedAllRequiredItems)
                     {
                         CollectionStatusText = Language.CollectionDownloadViewModel_Ready_to_install;
                     }
                     else
                     {
-                        CollectionStatusText = string.Format(Language.CollectionDownloadViewModel_Num_required_mods_downloaded, numDownloaded, RequiredDownloadsCount);
+                        CollectionStatusText = string.Format(Language.CollectionDownloadViewModel_Num_required_mods_downloaded, numDownloadedRequiredItems, RequiredDownloadsCount);
                     }
                 }).AddTo(disposables);
 
+            numDownloadedOptionalItemsObservable.CombineLatest(isPremiumObservable)
+                .OnUI()
+                .Subscribe(tuple =>
+                {
+                    var (numDownloadedOptionalItems, isPremium) = tuple;
+                    var hasDownloadedAllOptionalItems = numDownloadedOptionalItems == OptionalDownloadsCount;
+
+                    CountDownloadedOptionalItems = numDownloadedOptionalItems;
+                    _canInstallOptionalItems.OnNext(numDownloadedOptionalItems > 0);
+                    _canDownloadOptionalItems.OnNext(!hasDownloadedAllOptionalItems && isPremium);
+                }).AddTo(disposables);
```

Comment:
Seems this is never executed, and the initial values for the canExecute are all false